index.js

```diff
@@ -0,0 +1,99 @@
+const url = 'https://trektravel.herokuapp.com/trips';
+
+const successCallback = function successCallback(response) {
+  // console.log('POST request was successful');
+  // console.log(response);
+
+  let generatedHMTL = '<p>Everything went great,';
+  generatedHMTL += ` and your reservation <span id="res-name">${response.name}</span> has been added to the DB!</p>`;
+  $('#ajax-results').html(generatedHMTL);
+};
+
+$(document).ready(function() {
+  const getTrips = getTrips => {
+    $.get(url,
+      function (response) {
+        $('#trips ul').html('')
+        response.forEach((trip) => {
+          // let tripName = trip['name']
+          // console.log(tripName);
+
+          $('#trips ul').append(`<li><a href="" data-id="${trip.id}" >${trip.name}</a></li>`)
+        })
+      } // end function (response)
+    ) // end $.get
+    .fail(function(response){
+      console.log(response);
+      $('#fail').html('<p>Request was unsuccessful</p>');
+    });
+  }; // end getTrips()
+
+  const trip = function trip (id) {
+
+    let url = 'https://trektravel.herokuapp.com/trips/' + `${id}`
+
+
+    $.get(url, function(response) {
+
+      console.log('success');
+      console.log(response['name']);
+
+      // tripId = `${response.id}`
+
+      let tripInfo = `
+      <h2 data-id=${response.id}> ${response.name} </h2>
+      `;
+
+
+      $('#name').html(tripInfo)
+      $('#category').html(response.category)
+      $('#duration').html(response.duration)
+      $('#cost').html('$' + (response.cost).toFixed(2))
+      $('#about').html(response.about)
+
+      // I am not sure if putting the reservation code with in the scope of the $.get trip is best practice. It is here because this was the simplest way to access the trip's id. I tried using the .data()
```

Comment:
This is a good observation, both that putting this complicated success callback inline feels a little dirty, and that it's the simplest way to build a closure around the trip data.

Another way would be to write a function that builds the submit handler as a closure, something like:

```javascript
const buildReservationSubmitHandler = function(tripId) {
  const handler = function(event) {
    // your current function definition goes here
  };
  return handler;
};
```

This is a little more complex, but much better organized.